Doc/library/os.path.rst

```diff
@@ -488,6 +488,40 @@ the :mod:`glob` module.)
       Accepts a :term:`path-like object`.
 
 
+.. function:: splitroot(path)
+
+   Split the pathname *path* into a 3-item tuple ``(drive, root, tail)`` where
+   *drive* is a device name or mount point, *root* is a string of separators
+   after the drive, and *tail* is everything after the root. Any of these
+   items may be the empty string. In all cases, ``drive + root + tail`` will
+   be the same as *path*.
+
+   On POSIX systems, *drive* is always empty. The *root* may be empty (if *path* is
+   relative), a single forward slash (if *path* is absolute), or two forward slashes
+   (implementation-defined per `IEEE Std 1003.1-2017; 4.13 Pathname Resolution
+   <https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/V1_chap04.html#tag_04_13>`_.)
+   For example::
+
+      >>> splitroot('/home/sam')
+      ('', '/', 'home/sam')
+      >>> splitroot('//home/sam')
+      ('', '//', 'home/sam')
+      >>> splitroot('///home/sam')
+      ('', '/', '//home/sam')
+
+   On Windows, *drive* may be empty, a drive-letter name, a UNC share, or a device
+   name. The *root* may be empty, a forward slash, or a backward slash. For
+   example::
+
+      >>> splitroot('C:/Users/Sam')
+      ('C:', '/', 'Users/Sam')
+      >>> splitroot('//Server/Share/Users/Sam')
+      ('//Server/Share', '/', 'Users/Sam')
```

Comment:
Maybe some of these examples should use `bytes` instead of strings, since the function works with `bytes` just as well.

Also, does the function work with `os.PathLike` objects? If so, should we state that explicitly? If not, should it?

Reply:
Support for `bytes` and `os.PathLike` objects is mentioned in the first paragraph of the `os.path` documentation.

> This module implements some useful functions on pathnames. To read or write files see [open()](https://docs.python.org/3/library/functions.html#open), and for accessing the filesystem see the [os](https://docs.python.org/3/library/os.html#module-os) module. The path parameters can be passed as strings, or bytes, or any object implementing the [os.PathLike](https://docs.python.org/3/library/os.html#os.PathLike) protocol.

Reply:
None of the existing examples in this module use `bytes` examples, but I'd still be tempted to have one or two of the examples here use `bytes` paths. It can't hurt, in my opinion:

```suggestion
      >>> splitroot('/home/sam')
      ('', '/', 'home/sam')
      >>> splitroot('//home/sam')
      ('', '//', 'home/sam')
      >>> splitroot(b'///home/sam')
      (b'', b'/', b'//home/sam')

   On Windows, *drive* may be empty, a drive-letter name, a UNC share, or a device
   name. The *root* may be empty, a forward slash, or a backward slash. For
   example::

      >>> splitroot('C:/Users/Sam')
      ('C:', '/', 'Users/Sam')
      >>> splitroot('//Server/Share/Users/Sam')
      (b'//Server/Share', b'/', b'Users/Sam')
```

Reply:
Byte paths are a niche use case on POSIX. They're used because POSIX pathnames aren't guaranteed to conform to any particular encoding, and so highly portable applications will use `bytes` to avoid guessing.

I'm not aware of any use case for them on Windows. AFAIK they're supported only for conformity with POSIX.

I think we'd be drawing undeserved attention to them if we include them in the examples. It also makes the differences between the examples harder to grok, I think.

Reply:
Alright, let's leave this then!

Reply:
Hope that's alright! Thanks.